components/builder-core/src/file_walker.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::{Path, PathBuf};
+use walkdir::{WalkDir, Iter};
+use hab_core::package::{FromArchive, PackageArchive};
+use protocol::depotsrv;
+
+pub struct FileWalker {
+    walker: Iter,
+}
+
+impl FileWalker {
+    pub fn new(path: &str) -> Self {
+        FileWalker { walker: WalkDir::new(PathBuf::from(path)).follow_links(false).into_iter() }
+    }
+}
+
+pub fn extract_package(path: &Path) -> Option<depotsrv::Package> {
```

Comment:
I'd recommend changing this function signature to use `AsRef<Path>` 

```
pub fn extract_package<T: AsRef<Path>>(path: T) -> Option<depotsrv::Package> {}
```

The reasoning here is that all of the String and Path types in the standard library implement `AsRef<Path>` which make calling this public API function much more ergonomic! I wish I had done this in all the places where I made people pass me a &Path before ;)